src/site/content/en/fast/codelab-adapt-video-to-image-serving-based-on-network-quality/index.md

```diff
@@ -0,0 +1,210 @@
+  const video = document.getElementById('coverVideo');
+  const videoSource = video.getAttribute('data-src');
+  video.setAttribute('src', videoSource);
+  video.setAttribute('style', 'height: 100%; width: 100%; display:inline');
+} else {
+  // image loading code
+}
+```
+
+Here’s how the video loading code works: the video source is specified inside
+[data-src attribute](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes)
+in `index.html` and then lazy-loaded in `script.js`.
+
+```html
+<video id="coverVideo" autoplay loop muted data-src="https://cdn.glitch.com/b6491350-b058-4eb6-aa6c-55c93122073e%2FMatrix%2C%20Console%2C%20Hacking%2C%20Code.mp4?1551464245607"></video>
+```
+
+Because the video source is specified in the `data-src` attribute, it isn’t
+displayed or downloaded initially.
+[Data attributes](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes)
+allow you to store extra information on standard HTML elements. They can be
+named anything, as long as it starts with "data-".
+
+To actually display the video on the page, the location from `data-src` needs to
+be set as the `src` attribute of the video element. This is done in `script.js`.
+
+First, get the DOM element that contains the asset:
+
+```js
+const video = document.getElementById('coverVideo');
+```
+
+Then get the resource location from the `data-src` attribute:
+
+```js
+const videoSource = video.getAttribute('data-src');
+```
+
+And finally set that as the `src` attribute of the video element:
+
+```js
+video.setAttribute('src', videoSource);
+```
+
+The last line takes care of CSS positioning:
+
+```js
+video.setAttribute('style', 'height: 100%; width: 100%; display:inline');
+```
+
+## Step 3: Load image
+
+To conditionally load an image on slower networks, use the same strategy as for
+the video.
+
+Add an image element to `index.html` (right after the video element), and use
+the `data-src` attribute instead of the `src` attribute.
+
+```html
+<img id="coverImage" data-src="https://cdn.glitch.com/36529535-5976-40f8-979b-40c898b86bd0%2F36529535-5976-40f8-979b-40c898b86bd0_1_Sn80dgiwpMjBVrqjfiDbnA.jpg?1553003835358" />
+```
+
+In `script.js`, add code to set the image’s `src` attribute depending on the
+`effectiveType` of the network.
+
+```js
+if (navigator.connection.effectiveType === '4g') {
+  const video = document.getElementById('coverVideo');
+  const videoSource = video.getAttribute('data-src');
+  video.setAttribute('src', videoSource);
+
+  video.setAttribute('style', 'height: 100%; width: 100%; display:inline');
+} else {
+  const image = document.getElementById('coverImage');
+  const imageSource = image.getAttribute('data-src');
+  image.setAttribute('src', imageSource);
+
+  image.setAttribute('style', 'height: 100%; width: 100%; display:inline');
+}
+```
+
+## Try it out
+
+To test it yourself:
+
+1. Mouse over the editor, press the **App** button, then press the **Show**
```

Comment:
"press the **App** button  ... " can be shortened to "click **App**, then click **Show** to preview the app."